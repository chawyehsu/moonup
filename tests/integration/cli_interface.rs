use std::process::Command;

use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};

const BIN: &str = "moonup";

fn cli() -> Command {
    Command::new(get_cargo_bin(BIN))
}

#[test]
fn test_help() {
    assert_cmd_snapshot!(cli().arg("--help"));
    assert_cmd_snapshot!("default", cli().arg("default").arg("--help"));
    assert_cmd_snapshot!("install", cli().arg("install").arg("--help"));
    assert_cmd_snapshot!("pin", cli().arg("pin").arg("--help"));
    assert_cmd_snapshot!("run", cli().arg("run").arg("--help"));
    assert_cmd_snapshot!("show", cli().arg("show").arg("--help"));
    assert_cmd_snapshot!("update", cli().arg("update").arg("--help"));
    assert_cmd_snapshot!("which", cli().arg("which").arg("--help"));
}

#[test]
fn test_cmd_alias() {
    assert_cmd_snapshot!("install_alias", cli().arg("i").arg("--help"));
    assert_cmd_snapshot!("show_alias", cli().arg("list").arg("--help"));
    assert_cmd_snapshot!("show_alias", cli().arg("ls").arg("--help"));
    assert_cmd_snapshot!("update_alias", cli().arg("u").arg("--help"));
}
