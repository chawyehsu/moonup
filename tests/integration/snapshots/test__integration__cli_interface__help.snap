---
source: tests/integration/cli_interface.rs
info:
  program: moonup
  args:
    - "--help"
---
success: true
exit_code: 0
----- stdout -----
Moonup is a tool to manage multiple MoonBit installations.

If you find any bugs or have a feature request, please open an issue on
GitHub: https://github.com/chawyehsu/moonup/issues

Usage: moonup [OPTIONS] <COMMAND>

Commands:
  completions  Generate shell completions
  default      Set the default toolchain
  install      Install or update a MoonBit toolchain [aliases: i]
  pin          Pin the MoonBit toolchain to a specific version
  run          Run a command with a specific toolchain
  self-update  Update Moonup to the latest version
  show         Show installed and active toolchains
  update       Update MoonBit toolchains [aliases: u]
  which        Show the actual binary that will be run for a given command
  help         Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose...  Increase logging verbosity
  -q, --quiet...    Decrease logging verbosity
  -h, --help        Print help
  -V, --version     Print version

----- stderr -----
